include/boost/spirit/home/x3/char/detail/cast_char.hpp

```diff
@@ -22,7 +22,10 @@ namespace boost { namespace spirit { namespace x3 { namespace detail
     // The trick is to cast to an unsigned version of the source char first
     // before casting to the target. {P.S. Don't worry about the code, the
     // optimizer will optimize the if-else branches}
-
+#if BOOST_WORKAROUND(BOOST_MSVC, >= 1400)
```

Comment:
Please, either include boost workaround header or do not use it. Also, boost is not tested/supported under msvc < 16 for a while already, so you can just check for msvc compiler, i.e. just `ifdef _MSC_VER `.

Reply:
I included the relevant header. I kept BOOST_WORKAROUND because that's how this kind of thing is done in all other Spirit headers.